### ISSUE TITLE
KOSMAPOSLReconstruction optimisations

### ISSUE THREAD

**Issue by KrisThielemans** [UCL/STIR#1147]
# main culprit 
according to `callgrind` about half the computation time of `KOSMAPOSLReconstruction<TargetT>::compute_kernelised_image` is in
https://github.com/UCL/STIR/blob/21c38de27984290dae8c151f33faa655e0d1fd83/src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx#L99
I guess any function with `exp`-like features would work. If we stick to `exp` we could try a look-up table (as was done for `erf`)

# a tiny opportunity 
https://github.com/UCL/STIR/blob/21c38de27984290dae8c151f33faa655e0d1fd83/src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx#L836-L839
seems this can be outside the dz/dy/dx loops. It is also repeated at
https://github.com/UCL/STIR/blob/21c38de27984290dae8c151f33faa655e0d1fd83/src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx#L873-L874

**Comment by KrisThielemans** [comment]
@danieldeidda  what do you think?

**Comment by danieldeidda** [comment]
yes we can definitely move that outside the inner loop. However I don't follow the first comment. not sure what was done for erf
